In [ ]:
# Competition-Solution/notebooks/train/00_data_profiling/01_data_profiling_tweet_lengths.ipynb

---

##### <b>Imports</b>

In [2]:
import sys
from rich.console import Console

sys.path.append("../../../src")
import data_utils

console = Console()

##### <b>Loading the data</b>

In [3]:
df_call2action, df_fdgo, df_violence = data_utils.load_competition_data(data_dir="../../../data/raw/", type="train")

Loading competition data...

Data of type 'train' loaded successfully.

##### <b>Profiling the Length of the Tweets</b>

In [4]:
print(f"Subtask 1 (Call2Action):")
print(df_call2action["description"].str.len().describe())
print(f"Subtask 2 (DBO):")
print(df_fdgo["description"].str.len().describe())
print(f"Subtask 3 (Violence):")
print(df_violence["description"].str.len().describe())

Subtask 1 (Call2Action):
count     6840.000000
mean       158.426754
std        286.136028
min          1.000000
25%         43.000000
50%         88.000000
75%        180.000000
max      10000.000000
Name: description, dtype: float64
Subtask 2 (DBO):
count    7454.000000
mean      150.157231
std       287.985052
min         1.000000
25%        43.000000
50%        84.000000
75%       170.000000
max      7624.000000
Name: description, dtype: float64
Subtask 3 (Violence):
count     7783.00000
mean       161.50045
std        289.71872
min          1.00000
25%         45.00000
50%         89.00000
75%        180.00000
max      10000.00000
Name: description, dtype: float64


<u><p>Interpretation of Tweet Lengths (`description` column character length)</p></u>

*   **`count` (Number of Tweets with Length Calculated):**
    *   **C2A:** 6,840 tweets.
    *   **DBO:** 7,454 tweets.
    *   **VIO:** 7,783 tweets.

*   **`mean` (Average Character Length of Tweets):**
    *   **C2A:** ~158.4 characters on average.
    *   **DBO:** ~150.2 characters on average.
    *   **VIO:** ~161.5 characters on average.

*   **`std` (Standard Deviation of Tweet Character Lengths):**
    *   **C2A:** ~286.1 characters.
    *   **DBO:** ~288.0 characters.
    *   **VIO:** ~289.7 characters.
    *   *Observation:* Higher standard deviations compared to trial dataset (~217-235)

*   **`min` (Minimum Character Length of a Tweet):**
    *   **C2A:** Shortest tweet is 1 character long.
    *   **DBO:** Shortest tweet is 1 character long.
    *   **VIO:** Shortest tweet is 1 character long.

*   **`25%` (1st Quartile - 25th Percentile):**
    *   **C2A:** 25% of tweets are 43 characters or shorter.
    *   **DBO:** 25% of tweets are 43 characters or shorter.
    *   **VIO:** 25% of tweets are 45 characters or shorter.

*   **`50%` (Median - 2nd Quartile - 50th Percentile):**
    *   **C2A:** Half the tweets are 88 characters or shorter.
    *   **DBO:** Half the tweets are 84 characters or shorter.
    *   **VIO:** Half the tweets are 89 characters or shorter.

*   **`75%` (3rd Quartile - 75th Percentile):**
    *   **C2A:** 75% of tweets are 180 characters or shorter.
    *   **DBO:** 75% of tweets are 170 characters or shorter.
    *   **VIO:** 75% of tweets are 180 characters or shorter.

*   **`max` (Maximum Character Length of a Tweet):**
    *   **C2A:** Longest tweet is 10,000 characters.
    *   **DBO:** Longest tweet is 7,624 characters.
    *   **VIO:** Longest tweet is 10,000 characters.

<u><p>Summary & Implication of Tweet Lengths</p></u>

*   *Observation:* The training dataset shows similar central tendencies to the trial dataset (median ~84-89 characters) but with higher variability and more extreme outliers. This matches what we saw in `00_data_profiling_basic.ipynb`, when we checked the overlapping tweets.
*   *Implication:*
    *   **Outlier Investigation:** The extremely long tweets (7,624-10,000 characters) will need to be investigated to determine if they represent valid content or data artifacts that we need to filter.
    *   **Model Considerations:** As outlined during our trial dataset profiling, these extremely long tweets will need to be taken into consideration. Either by deciding if they should be kept at all, truncated, the respective approach chosen, or model and thereby architecture chosen.
    *   **Preprocessing Strategy:** The wider tweeet-length distribution suggests that length-based filtering or normalization may be more critical for the training dataset than initially thought based on the trial data. Good thing that we are also running EDA on the training set! :=D

##### Profiling the shortest and longest tweets

In [5]:
for df, subtask_name in zip([df_call2action, df_fdgo, df_violence], ["Call2Action", "DBO", "Violence"]):
    console.print(f"Subtask: {subtask_name}", style="bold")
    shortest_tweets_series, longest_tweets_series = data_utils.get_shortest_longest_tweets(
        dataframe=df,
        column="description",
        top_n=3,
        short_length_range=3,
        long_length_range=500
    )
    
    console.line(2)

Subtask: Call2Action

Getting up to 3 tweets from specified length ranges (Short range span: 3, Long range span: 500)...

--- Displaying up to 3 Shortest Tweets (Lengths from 1 to 3) ---

Index: 3422, Length: 1

Text: "S"

Index: 6544, Length: 1

Text: "9"

Index: 6666, Length: 1

Text: "?"

--- Displaying up to 3 Longest Tweets (Lengths from 9501 to 10000) ---

Index: 2419, Length: 10000

Text: "FAKT IST! dass für alle migranten in deutschland die selben ausgangs- und startbedingen gelten. die soziale 
entwicklung und integration ist hingegen stark abhängig von deren herkunftsland und ihrem kulturellen hintergrund. 
polen, tschechen, chinesen, vitnamesen besitzen im vergleich zu anderen migrationsgruppen eine bessere bilanz der 
erfolgreichen integration.>>  http://www.sueddeutsche.de/.../migranten-in-deutschland ... <<FAKT IST! wer sich 
nicht integrieren will, integriert sich nie. man kann keinen menschen durch gesetze zur integration zwingen, wenn 
die innere bereitschaft dazu fehlt.FAKT IST! dass in berlin neukölln die arbeitslosigkeit türkisch stämmiger 
einwohner bei ca. 50% liegt und dass ca. 75% der türkischen schüler die schule ohne einen schulabschluss 
verlässt.>>  https://www.youtube.com/watch?v=qz3chLW-128 ... <<FAKT IST! dass es sich deutschland nicht leisten 
kann, in der immer stärker werdenden globalisierten, spezialisierten, technisierten welt, einwanderungsgruppen zu 
verlieren. >>  https://sioede.wordpress.com/tag/schulschwache-migranten/  <<FAKT IST! dem fachkräftemangel und 
demografischen problem mit ungesteuerter zuwanderung entgegenzutreten, ist keine lösung. es gibt genügend regionen,
wo fachkräfte den mangel suchen. für nicht jedes fachgebiet besteht in deutschland eine knappheit an spezialisten. 
in den fachgebieten, wo bedarf besteht, kann mit gezielter anwerbung abhilfe getan werden. wir sollten uns dabei 
immer im hinterkopf behalten, dass wir den strukturschwachen ländern ihres menschlichen potentials und ihrer 
fachkräfte berauben! sollten wir nicht erst einmal den deutschen menschen, die nach einer neuen perspektive suchen,
helfen fachkräfte zu werden?FAKT IST! deutschland wird von den politikern zum einwanderungsland deklariert und das 
volk muss mit ihnen leben. >>  http://www.merkur-online.de/.../wulff-deutschland ... <<FAKT IST! integration 
erfordert 100 prozentigen einsatz, ohne sprache, einem qualifizierten und in deutschland anerkannten beruf ist eine
wirtschaftliche und soziale teilhabe ausgeschlossen. viele kriegsflüchtlinge aus den islamischen oder afrikanischen
ländern (vor allem die frauen) besitzen eine schlechte bzw. eine unzureichende schulausbildung und/oder keine bzw. 
eine nicht den deutschen standards entsprechende berufsausbildung. viele schaffen den beruflichen einstieg in ihr 
neues leben nicht und werden lebenslang die sozialkassen und damit den steuerzahler belasten.>>  
http://www.bamf.de/.../Wor.../wp13-schulische-bildung.pdf ... <<FAKT IST! nicht alle arabischen, 
nordamerikanischen, pazifischen und europäischen länder sind zur aufnahme von kriegsflüchtlingen (in dem ausmaß) 
wie deutschland bereit. (polen, tschechien, australien, usa, kanada, viele golfstaaten) wo sonst, als in den 
reichen golfstaaten, wäre integration, bei gleicheit von kultur, religion, schrift (semiotik), (sprache) und 
lebensweise, so leicht möglich, als in diesen ländern. im koran wird ausdrücklich zur gegenseitigen solidarität der
muslime dieser welt aufgerufen.>>  http://www.aargauerzeitung.ch/.../oelstaaten-wollen-keine ... <<FAKT IST! dass 
mit den kosten eines kriegsflüchtlings in deutschland in den flüchtlingscamps (vor ort) ganze großfamilien versorgt
werden könnten. FAKT IST! bestimmte zuwanderergruppen fallen europaweit durch bildungsferne, kriminalität und 
arbeitslosigkeit auf. hierbei von einem versagen euopäischer einwanderungspolitik zu sprechen wäre unpassend, da 
diese symptome ein merkmal spezieller migrantengruppen sind. >>  
http://www.deutschlandfunk.de/schweden-die-integration ...  http://europenews.dk/de/node/50924  <<FAKT IST! wir 
brauchen keine asylanten, die hier ihre lebensweise fortführen, welche in ihren ländern zu den zuständen geführt 
hat, vor welchen sie geflohen sind.FAKT IST! in deutschland haben sich kriminelle clans und großfamilien etabliert,
denen die polizei und die justiz nicht mehr gewachsen sind. zeugen werden zum schweigen gebracht, 

Finished getting tweets from specified length ranges.

Subtask: DBO

Getting up to 3 tweets from specified length ranges (Short range span: 3, Long range span: 500)...

--- Displaying up to 3 Shortest Tweets (Lengths from 1 to 3) ---

Index: 1012, Length: 1

Text: "H"

Index: 1835, Length: 1

Text: "B"

Index: 2790, Length: 2

Text: "??"

--- Displaying up to 3 Longest Tweets (Lengths from 7125 to 7624) ---

Index: 2183, Length: 7624

Text: "22. November 2015, liebe Leser, liebe Zornbürger, liebes Pack, liebe Idioten,

Frau Merkel wurde am Parteitag der CSU zusammengefaltet, erstaunlich, wie leicht das geht?!

Wir durften hier teilhaftig werden an einem "staatsfraulichen" Auftritt der "mächtigsten" Frau der Welt, wie es 
wohl jämmerlicher nicht mehr möglich ist!
Ja, und es ist geradezu faszinierend, daß sich dieses Getüm noch immer als Kanzlerin behaupten kann?!
Selbst die "Huffington Post" fand heute ein paar Worte zu Frau Merkel, der Chefredakteur Sebastian Matthes schrieb 
unter dem Titel:

"Liebe Frau Merkel, es ist Zeit für die Wahrheit" einige passende oder auch vielmehr unpassende Zeilen:

Passende Zeilen deshalb, weil er durchaus Kritik am Merkelschen Treiben äußerte, unpassend deshalb, weil er das 
Merkelsche Treiben lediglich einem gewissen "Fehlverhalten, einer gewissen Unkenntnis und Ignoranz" zuordnete.

Falsch, Herr Matthes, das, was Frau Merkel treibt, ist nichts anderes als der sehr wohl klar und immer klarer 
erkennbar betriebene Auftrag, dieses Land auftragsgemäß zu zerstören; wenn Sie, Herr Matthes, Frau Merkel u.a. 
"Kenntnis und Vorausschau" absprechen, dann sollten Sie sich folglich einmal mit dieser Variante einer Erklärung 
näher beschäftigen, Sie werden Erstaunliches vorfinden!

Eine junge Syrerin, angeblich Studentin der Medizin, kam heute im Radio zu Wort, sie sagte:

"Wir sind verzweifelt, in Syrien hat man uns gesagt, daß jede Flüchtlingsfamilie in Deutschland ein Haus bekommt, 
jede Person zwischen 3.000 - 5.000 Euro Startkapital und nun das hier, wir sitzen in Lagern und wissen nicht 
weiter..."

Wieder einmal habe ich ein gewisses Problem mit einer "Tatsache", welche um 180 Grad zur Wahrheit 
auseinanderdriftet:

Angeblich qualifizierte Flüchtlinge, potentielle Facharbeiter, alle Flüchtlinge ausgestattet mit schmartfon ( und 
äpps ), bestens informiert über "Flucht"-Routen und sonstige Verhältnisse, genau diese kommen plötzlich "völlig 
desinformiert" und von den Zuständen hier (Aufnahmelager, Chaos etc.) "überrollt" hier an, wer soll das denn 
glauben??

Das glaubt niemand, der seinen gesunden Verstand beisammen hat, denn all das dient nur einem einzigen Zweck, dem 
Zweck der weiteren Schuld der Deutschen, denn wenn sich diese Menschen, nun hier im Land, wiederum als "Opfer von 
Lügen" gerieren können, dann ist ihnen erneut das Mitleiden der Gutmenschen sicher!
Oder auch nicht oder auch nicht mehr lange?!

Bleiben wir bei der "Huffington Post", diesmal eine SchreiberIn (das I ist eine homage an die Grünen), Frau Sabrina
Hoffman, Chefin vom Dienst.
Sie stellt fest:

"Das Biest ist entfesselt: Die Herrschaft der Aggro-Deutschen"

Sie stellt in ihrem Beitrag u.a. fest, daß die Deutschen nach dem Terror in Paris "aggressiv" reagieren, während 
die durch Terror vielmehr belasteten Franzosen sich eher auf "Frieden und Zusammenhalt" besinnen.

Das ist schön, wir sollten es also genauso halten mit Frau Merkel, die unlängst in der Schweiz die Bürger 
aufforderte, als Gegenmittel gegen Terror und Islamisierung mehr in die Kirche zu gehen!

Ja, das hätten auch die Pariser tun sollen, dann hätte es nicht 132 Tote gegeben, nicht wahr?

Liebe Frau Hoffmann, von was Sie da in Ihrem Beitrag schwadronieren, das zeugt von erheblicher Schmerzresistenz!
Sie vergleichen die Situation Frankreichs und die Haltung der Bürger dort mit Deutschland und seinen Bürgern??

Ich darf gelinde darauf aufmerksam machen, daß die Situation Deutschlands derzeit einmalig in Europa, vielleicht in
der Welt ist!
Während hier ungezählt, unregistriert oder auch registriert, Hunderttausende ins Land strömen, gleichzeitig Städte 
und Kommunen, freiwillige Helfer und Ordnungskräfte, die Polizei und Sicherheitsdienste am Boden liegen, der 
finanzielle Kollaps der Städte und Gemeinden bereits eingetreten ist, Lehrer in Schulen Alarm schlagen, in 
Krankenhäusern das Personal bereits von Sicherheitsdiensten beschützt werden muß (letzte Meldung aus Sigmaringen, 
Kosten für di

Index: 4304, Length: 7433

Text: "Frankfurt

Da kommt dein Flugplatz noch dazu ,einfach keinen Mist erzählen!

So viele Verbrechen wie lange nicht
2014 war in Frankfurt ein Jahr mit vielen Morden – die meisten sind geklärt. Ein Überblick über ein nicht „ganz 
normales“ Jahr für die Ermittler.

Jahreswechsel mit einem Verbrechen beginnt, kommt nicht selten vor – nach Auskunft von Kriminalpsychologen sogar 
ziemlich häufig. Denn Weihnachten, so sagen sie, sei eine „hochemotionale Zeit“. Viele Familien seien sich an den 
Festtagen so nah wie selten, komme es zum Streit, sei die Gefahr hoch, dass die Situation eskaliere.

Dieses Weihnachtsfest ist für die Frankfurter Polizei so gesehen fast schon eine Ausnahme gewesen. Denn zum 
Ausklang des Jahres ist es in der Mainmetropole ungewöhnlich ruhig geblieben – ganz anders als in den zwölf Monaten
zuvor. Aus Sicht der Ermittler ist das Jahr 2014 nämlich kein „ganz normales“ gewesen. Nur zwei Tage nach Neujahr 
ereignete sich der erste Mord – und der sollte nur der Auftakt zu einer ganzen Reihe von Tötungsdelikten sein. Das 
ganze Jahr über wurde in Frankfurt und Umgebung geschossen, erstochen und anderweitig getötet. So viel wie lange 
nicht mehr.

Mysteriöser Kopfschuss-Mord
Eines der Verbrechen, das die Polizei bis heute vor ein Rätsel stellt, war der Mord an dem Wettspielbetreiber 
Oliver F. am 3. Januar 2014. Am helllichten Tage wurde der Mann in Frankfurt-Sachsenhausen im Innenhof des 
Gebäudes, in dem er sein Büro hatte, erschossen. Nach bisherigen Erkenntnissen lauerte der Täter dem Geschäftsmann 
auf und schoss ihm zwei Kugeln in den Kopf. Es gab keine Zeugen und keine brauchbaren Aufzeichnungen von 
Überwachungskameras. Die offizielle Stellungnahme der Polizei lautet bis heute, sie ermittle „in alle Richtungen“. 
Allerdings, so ist aus dem Kreis der Ermittler zu hören, wird ein geschäftliches Motiv vermutet. F., der das 
Unternehmen Happybet zusammen mit seinem Bruder aufgebaut hatte, soll Gerüchten zufolge geplant haben, aus 
bestehenden Verträgen mit ausländischen Wettanbietern auszusteigen, um unabhängig zu sein. Nach Ansicht der 
Ermittler ging es möglicherweise um viel Geld. Selbst wenn der eigentliche Schütze jemals gefasst werden sollte, 
sei fraglich, ob die Hintermänner ermittelt werden könnten, glauben die Beamten. Denn nach allem, was die 
Auswertung der Spuren bisher ergeben hat, handelte es sich um einen Auftragsmord.

Weit weniger rätselhaft ist der Mord an zwei Afghanen vor dem Frankfurter Landgericht am 24. Januar 2014. Der 
Prozess gegen den mutmaßlichen Täter, den 48 Jahre alten Afghanen Hamayon S., wird gerade vor dem Landgericht 
verhandelt. Das Motiv scheint klar: Kurz nach der Tat sprach der heutige Frankfurter Polizeipräsident Gerhard 
Bereswill von „Blutrache“, einer Form der Selbstjustiz mit kulturellen Motiven.

Rache - ein häufiges Motiv
Auch der Prozess zeigt: Die Tat steht in engem Zusammenhang mit einem Verbrechen, das sich bereits vor sieben 
Jahren zugetragen hat, auf einem Parkplatz in Steinbach. Damals ging es um einen Streit zwischen Autohändlern, bei 
dem der Bruder von Hamayon S. getötet wurde – mutmaßlich von den beiden Männern, die S. schließlich vor dem 
Landgericht erschoss. Beide wurden in einem ersten Urteil freigesprochen. Hamayon S., so stellt sich die Tat heute 
dar, wollte seinen toten Bruder rächen. Die Frankfurter Polizei befürchtet weitere Racheakte zwischen den beiden 
afghanischen Familien – auch wenn es derzeit keine konkreten Hinweise darauf gibt.

Die Morde vor dem Landgericht blieben nicht die einzige Tat, die von Rache getrieben war. Denn auch die tödlichen 
Schüsse, die im April 2014 im Frankfurter Stadtteil Nieder-Eschbach fielen, waren nach derzeitigen Erkenntnissen 
die Reaktion auf einen Streit, der schon länger zwischen zwei jungen Männern schwelte. Angeblich ging es um 
Drogengeschäfte. Der Fall des toten Kibrom T. beschäftigte die Polizei über Wochen hinweg. Auch deshalb, weil T. 
Mitglied der Rockervereinigung MC Gremium Fulda war und die

Index: 7332, Length: 7152

Text: "Bitte kopiert den Text, wenn er Euch gefällt und sendet diesen Text ebenfalls an alle Pegida-, Legida-, 
Pegada-, Endgame-, Monatgsdemo- und Mahnwachenseiten, die es gibt, Nur wenn wir uns zusammenschließen und die 
Kernthemen benennen, werden wir Schlimmes im Land verhindern können !!!
.....
Pegida?
Legida?
Pegada?
Mahnwachen für den Frieden?
Montagsdemos?
LEGIDA?
Gegendemos?
Antifa?
Terrordrohungen gegen Bürgerbewegungen?
Demonstrationsverbote aufgrund zweifelhaften Twitter nachrichten?
Scheint alles ziemlich verwirrend zu sein!
Sind wir jetzt an der Schwelle zur "völligen Verwirrung und Abschaffung jeglicher Demokratie und des gesunden 
Menschenverstandes"?
Man demonstriert ist für oder gegen etwas.
Tausende Menschen treibt es auf die Straßen!
Es rumort im Lande. Viele Bürger fühlen sich nicht mehr durch die Politik im Land vertreten und durch die Medien 
richtig informiert.
Ich selbst habe auf Montagsmahnwachen die zahlreichen Missstände öffentlich dargestellt und kritisiert und sehe mit
geteilten Gefühlen auf all diese unterschiedlichen Bewegungen.
Es ist das altbekannte „TEILE UND Herrsche-Spiel, welches unübersehbar die Bevölkerung in konkurrierende Gruppen 
teilt. Dieses Teilen wird sogar medial noch geschürt.
Während die Montagsmahnwachen mit hohem intellektuellem Anspruch und vielseitigen Themen ( klar war beim freien 
Mikro auch vieles Grenzwertiges dabei ) von den Medien in unglaublichster Praxis einer real REAL EXISTIERENDEN 
ZENSUR schlichtweg ignoriert wurden, halten die Medien frontal auf die Pegida drauf und manipulieren sogar 
Interviews mit gekauften RTL-Reportern, die das ganze Geschehen in eine rechte Ecke rücken sollten.
Dass die Pegida das Geldsystem und die fehlende Souveränität in Deutschland konsequent auslässt, ist sehr schade, 
denn diese Zusammenhänge gehören dazu, auch wenn man diese dem gemeinen Stammtischpublikum zunächst nahe bringen 
müsste.
Dass immer dann eine Gegendemonstration von der gut konditionierten deutschen Masse mit erhobenem Zeigefinger und 
Unterstützung der Systemmedien und Politdarsteller stattfindet ist ja Langerprobte andressierte Praxis.
Interessant auch, dass seit dem 11.9. die westlichen Medien alles getan haben, um insbesondere moslemische Länder 
mit einem Stigma zu versehen. Nicht nur völkerrechtlich inakzeptable Kriege hat der Westen in diesen Ländern 
geführt, sondern auch fleißig Waffen geliefert und bezahlt.
Auch Al Kaida und ISIS sind letztlich Geburten der westlichen Imperialisten. Noch heute fliegen regelmäßig Drohnen 
und Bomben in diesen Ländern unschuldige Menschen ins Jenseits, ohne dass eine Klage, Verteidigung oder ein 
Schuldspruch erfolgt wäre. In Guantanamo werden noch immer Menschenrechte mit Füßen getreten und die Presse schaut 
aktiv weg! Selbst Die Kurden bekommen jetzt Waffen „GESCHENKT“, ein NOVUM! Direkte Waffenlieferungen in 
Krisengebiete sind verboten und inakzeptabel und machen uns ohne Kriegserklärung und UN Mandat mit zu einer 
Kriegspartei.
Und jetzt wo aus eben diesen Ländern Menschen fliehen und Schutz im Ausland – auch bei uns – suchen, stellt der 
Otto Normal Verbraucher fest, dass Muslime fliehen. Wer bitteschön soll denn sonst fliehen, wenn man Bürgerkriege 
in muslimischen Ländern führt und schürt?
Das Volk wurde medial ja schon so „dressiert“, dass es hinter jedem friedlichen Muslim einen potentiellen 
Terroristen vermutet. Einen Islamisten, mit Sprengstoffgürtel.
Das entspricht genauso wenig der Wahrheit, wie die „Volksmeinung“ die im 3. Reich gezielt gegen eine andere 
Religion „andressiert“ wurde.
Dabei spielt die Religion meiner Meinung nach eine untergeordnete Rolle.
Sollten die PEGIDA-Anhänger, die GEGENDEMONSTRANTEN, KRICHEN, DEMOKRATEN, MONTAGSDEMONSTRANEN UND 
MAHNWACHENTEILNEHMER wirklich nicht erkennen, dass unsere Feinde nicht die Flüchtlinge sind, sondern die Mächtigen 
auf den Regierungsbänken und in den Geheimdiensten, die unerklärte Kriege mit unserem Geld führen? Die Gesetze 
erlassen, die das einheimische Vo

Finished getting tweets from specified length ranges.

Subtask: Violence

Getting up to 3 tweets from specified length ranges (Short range span: 3, Long range span: 500)...

--- Displaying up to 3 Shortest Tweets (Lengths from 1 to 3) ---

Index: 2371, Length: 1

Text: "S"

Index: 6531, Length: 1

Text: "?"

Index: 6105, Length: 1

Text: "9"

--- Displaying up to 3 Longest Tweets (Lengths from 9501 to 10000) ---

Index: 6216, Length: 10000

Text: "FAKT IST! dass für alle migranten in deutschland die selben ausgangs- und startbedingen gelten. die soziale 
entwicklung und integration ist hingegen stark abhängig von deren herkunftsland und ihrem kulturellen hintergrund. 
polen, tschechen, chinesen, vitnamesen besitzen im vergleich zu anderen migrationsgruppen eine bessere bilanz der 
erfolgreichen integration.>>  http://www.sueddeutsche.de/.../migranten-in-deutschland ... <<FAKT IST! wer sich 
nicht integrieren will, integriert sich nie. man kann keinen menschen durch gesetze zur integration zwingen, wenn 
die innere bereitschaft dazu fehlt.FAKT IST! dass in berlin neukölln die arbeitslosigkeit türkisch stämmiger 
einwohner bei ca. 50% liegt und dass ca. 75% der türkischen schüler die schule ohne einen schulabschluss 
verlässt.>>  https://www.youtube.com/watch?v=qz3chLW-128 ... <<FAKT IST! dass es sich deutschland nicht leisten 
kann, in der immer stärker werdenden globalisierten, spezialisierten, technisierten welt, einwanderungsgruppen zu 
verlieren. >>  https://sioede.wordpress.com/tag/schulschwache-migranten/  <<FAKT IST! dem fachkräftemangel und 
demografischen problem mit ungesteuerter zuwanderung entgegenzutreten, ist keine lösung. es gibt genügend regionen,
wo fachkräfte den mangel suchen. für nicht jedes fachgebiet besteht in deutschland eine knappheit an spezialisten. 
in den fachgebieten, wo bedarf besteht, kann mit gezielter anwerbung abhilfe getan werden. wir sollten uns dabei 
immer im hinterkopf behalten, dass wir den strukturschwachen ländern ihres menschlichen potentials und ihrer 
fachkräfte berauben! sollten wir nicht erst einmal den deutschen menschen, die nach einer neuen perspektive suchen,
helfen fachkräfte zu werden?FAKT IST! deutschland wird von den politikern zum einwanderungsland deklariert und das 
volk muss mit ihnen leben. >>  http://www.merkur-online.de/.../wulff-deutschland ... <<FAKT IST! integration 
erfordert 100 prozentigen einsatz, ohne sprache, einem qualifizierten und in deutschland anerkannten beruf ist eine
wirtschaftliche und soziale teilhabe ausgeschlossen. viele kriegsflüchtlinge aus den islamischen oder afrikanischen
ländern (vor allem die frauen) besitzen eine schlechte bzw. eine unzureichende schulausbildung und/oder keine bzw. 
eine nicht den deutschen standards entsprechende berufsausbildung. viele schaffen den beruflichen einstieg in ihr 
neues leben nicht und werden lebenslang die sozialkassen und damit den steuerzahler belasten.>>  
http://www.bamf.de/.../Wor.../wp13-schulische-bildung.pdf ... <<FAKT IST! nicht alle arabischen, 
nordamerikanischen, pazifischen und europäischen länder sind zur aufnahme von kriegsflüchtlingen (in dem ausmaß) 
wie deutschland bereit. (polen, tschechien, australien, usa, kanada, viele golfstaaten) wo sonst, als in den 
reichen golfstaaten, wäre integration, bei gleicheit von kultur, religion, schrift (semiotik), (sprache) und 
lebensweise, so leicht möglich, als in diesen ländern. im koran wird ausdrücklich zur gegenseitigen solidarität der
muslime dieser welt aufgerufen.>>  http://www.aargauerzeitung.ch/.../oelstaaten-wollen-keine ... <<FAKT IST! dass 
mit den kosten eines kriegsflüchtlings in deutschland in den flüchtlingscamps (vor ort) ganze großfamilien versorgt
werden könnten. FAKT IST! bestimmte zuwanderergruppen fallen europaweit durch bildungsferne, kriminalität und 
arbeitslosigkeit auf. hierbei von einem versagen euopäischer einwanderungspolitik zu sprechen wäre unpassend, da 
diese symptome ein merkmal spezieller migrantengruppen sind. >>  
http://www.deutschlandfunk.de/schweden-die-integration ...  http://europenews.dk/de/node/50924  <<FAKT IST! wir 
brauchen keine asylanten, die hier ihre lebensweise fortführen, welche in ihren ländern zu den zuständen geführt 
hat, vor welchen sie geflohen sind.FAKT IST! in deutschland haben sich kriminelle clans und großfamilien etabliert,
denen die polizei und die justiz nicht mehr gewachsen sind. zeugen werden zum schweigen gebracht, 

Finished getting tweets from specified length ranges.